desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java

```diff
@@ -751,7 +751,7 @@ protected boolean onWindowClosing() {
 
                 if (unsavedChangesChoice == 0) { // save changes
                     _saveButton.doClick();
-                } else if (unsavedChangesChoice == 2) { // cancel closing
+                } else if (unsavedChangesChoice != 1) { // cancel closing
```

Comment:
I still believe you are making a mistake, sorry. Make a mental excercise which will (dis)prove your idea: What if Oracle developer decides to change the constant value? Normally, this shouldn't brake anything. But your code would be broken. Because the constant value in JOptionPane.YES_OPTION is totally unrelated to an index in our(!) array of options... You cannot use some constant in you comparison just because now it has - by coincidence - the same value as you need, but the semantic of the constant is totally different than the value you are comparing.